cloud/ec2.go

```diff
@@ -595,25 +595,22 @@ func (m *ec2Manager) setNoExpiration(ctx context.Context, h *host.Host, noExpira
 	}
 
 	if noExpiration {
-		instance, err := m.client.GetInstanceInfo(ctx, h.Id)
+		if err := h.MarkShouldNotExpire(ctx, expireOnValue); err != nil {
+			return errors.Wrapf(err, "marking host should not expire in DB for host '%s'", h.Id)
+		}
+
+		// Use GetInstanceStatus to add/update the cached host data (including
+		// unexpirable host information like persistent DNS names and IP
+		// addrseses) if the unexpirable host is running.
+		_, err := m.GetInstanceStatus(ctx, h)
 		grip.Error(message.WrapError(err, message.Fields{
-			"message":    "could not get instance info for assigning persistent DNS name",
+			"message":    "could not get instance info to assign persistent DNS name",
```

Comment:
We already log this info https://github.com/evergreen-ci/evergreen/blob/9d5894225f5bafadf31c63eeeb545e24d0beab92/cloud/ec2.go#L783 so we're double logging errors; if you want this for the dashboard could we log at a debug level instead, or just remove?

Reply:
I think just removing this log inside `GetInstanceStatus` is good enough, the caller will likely log it anyways.